src/pickup.cpp

```diff
@@ -753,16 +756,28 @@ void Pickup::pick_up( const tripoint &pos, int min )
 
             if( idx >= 0 && idx < (int)here.size()) {
                 if( getitem[idx] ) {
-                    if( getitem[idx].count != 0 && getitem[idx].count < here[idx].charges ) {
-                        item temp = here[idx];
-                        temp.charges = getitem[idx].count;
-                        new_weight -= temp.weight();
-                        new_volume -= temp.volume();
-                    } else {
-                        new_weight -= here[idx].weight();
-                        new_volume -= here[idx].volume();
+                    if(here[idx].count_by_charges()) {
```

Comment:
You're shifting away from the project coding style here, parentheses are separated from their contents by a space.
Also on line 762-763, else goes on the dame line as its matching bracket.